### ISSUE TITLE
Generate setter stubs for the Directives

### ISSUE THREAD

**Issue by mhevery** [angular/angular#500]
When transformer comes across a directive it needs to extract the setter stubs from the binding. 

Given this code:

```
@Decorator(
  selector: '[tool-tip]',
  bind: const {
    "tool-tip": "text"
  }
)
class TooltTip {
  String text;
}
```

From the above example the transformer needs to extract all of the `bind`ings and generate: 
```
reflector.registerSetter({
  "text": (TooltTip o, String value) => o.text = value
});
```

This allows angular to write changes from the change detection into the directive bindings.

NOTE: If there are more than one Directive which has the same setter, we may have to use nearest common supper class for the stub. (We can devise a more accurate mechanism later)

**Comment by kegluneq** [comment]
Per [mhevery comment on 780](https://github.com/angular/angular/pull/780), `bind` syntax has changed so the map is flipped in the annotation, that is:

`{"tool-tip": "text"} => {"text": "tool-tip"}`

**Comment by kegluneq** [comment]
Fixed in d0aceef4e02a220703a570dec0a8543051e7182f
